packages/react-native-editor/__device-tests__/pages/editor-page.js

```diff
@@ -258,31 +258,42 @@ class EditorPage {
 	// =========================
 
 	async addNewBlock( blockName, relativePosition ) {
-		// Click add button.
-		let identifier = 'Add block';
-		if ( isAndroid() ) {
-			identifier = 'Add block, Double tap to add a block';
-		}
-		const addButton = await this.driver.elementByAccessibilityId(
-			identifier
+		const addBlockButtonLocator = isAndroid()
+			? '//android.widget.Button[@content-desc="Add block, Double tap to add a block"]'
+			: '//XCUIElementTypeButton[@name="add-block-button"]';
+
+		const addButton = await waitForVisible(
+			this.driver,
+			addBlockButtonLocator
 		);
 
 		if ( relativePosition === 'before' ) {
 			await longPressMiddleOfElement( this.driver, addButton );
+			const addBlockBeforeButtonLocator = isAndroid()
+				? '//android.widget.Button[@content-desc="Add Block Before"]'
+				: '//XCUIElementTypeButton[@name="Add Block Before"]';
 
-			const addBlockBeforeButton = await this.driver.elementByAccessibilityId(
-				'Add Block Before'
+			const addBlockBeforeButton = await waitForVisible(
+				this.driver,
+				addBlockBeforeButtonLocator
 			);
-
 			await addBlockBeforeButton.click();
 		} else {
 			await addButton.click();
 		}
 
 		// Click on block of choice.
 		const blockButton = await this.findBlockButton( blockName );
+
 		if ( isAndroid() ) {
 			await blockButton.click();
+
+			// For certain blocks, clicking on it will display additional options that the test should wait for before next steps
+			if ( blockName === 'Image' ) {
```

Comment:
This seems to only be affecting Android tests, there could be more but for this test file, looks like Image block is one of them. The locator is the part highlighted below:

![image block](https://user-images.githubusercontent.com/17252150/160048605-c3c2b9fd-8174-483f-9c10-8de0bc954737.png)

Reply:
Media blocks like Image automatically open the media options picker. In this case, I think that waiting for the block picker should be part of the test that adds the block. Which by the way is already being done, as outlined in the following example:

https://github.com/WordPress/gutenberg/blob/a4c8730ced637818b02435b6a40e73f83051ed3e/packages/react-native-editor/__device-tests__/gutenberg-editor-image-%40canary.test.js#L10-L12

Although it would be great to use the `waitForVisible` helper for that element (i.e. media options picker) too 😅 .